### DISCUSSION TITLE
Saved tsdate runs for SGDP and 1000 Genomes?

### DISCUSSION THREAD

**Discussion by clawsoon** [tskit-dev/tskit#1311] [Q&A] [answered]
Hi,

I'm not sure if this is the right place to ask, but perhaps someone can send me in the right direction if it's not.

Does anyone have saved tsdate results for the tsinfer 1000 Genomes and SGDP tree sequences ([here](https://zenodo.org/record/3051855) and [here](https://zenodo.org/record/3052359)) that they could share for me to play around with?

I tried to run tsdate myself, but I only have 4GB on my Linux machine and it ran out of memory on chromosome 22.

Thanks.

**Comment by awohns** [comment, marked as answer]
Hi @clawsoon, thanks for your question. There isn't a publicly available, "tsdated" version of the 1000 Genomes and SGDP tree sequences which you link to. Those tree sequences were created using the original release of `tsinfer` from this [paper](https://www.nature.com/articles/s41588-019-0483-y). We will soon be releasing the dated tree sequences of the HGDP+1000 Genomes+SGDP+Four archaics+Afanasievo quartet from "[A unified genealogy of modern and ancient genomes](https://www.biorxiv.org/content/10.1101/2021.02.16.431497v1)". If you wanted just the 1000 Genomes or SGDP, you could [simplify](https://tskit.dev/tskit/docs/stable/python-api.html?highlight=simplify#tskit.TreeSequence.simplify) that tree sequence down to the samples you were interested in. 
Hope this is helpful!

**Reply by clawsoon** [reply]
Hi @awohns.  That sounds like exactly what I'm looking for.  I'll wait for the release.  Thanks!

**Reply by clawsoon** [reply]
EDIT: I'll move this to a new question.

Hi @awohns .  Did the HGDP+1000 Genomes+SGDP+Four archaics+Afanasievo dated tree sequences get released yet?

My attempt to run the code from your unified_genealogy_paper repo for chromosome 22 finally finished the other day after chewing away for a couple of months (and a few false starts).  The resulting dates in it seem a little odd, though, since the oldest time is 81,721 and the times for the ancients range from 184 to 4,400.  Am I guessing correctly that those are generation counts and I should be multiplying by 20 (or 27 or 28?) to get actual dates?

Thanks.

**Reply by awohns** [reply]
Hi @clawsoon, thanks for checking in on this! I'm in the process of validating the files and uploading them to Zenodo. They should be available by the end of this week.
And yes, you're correct that the dates are in generations and should be multiplied by 25.
Let me know if you have any other questions!

**Reply by clawsoon** [reply]
Perfect timing, @awohns.  I'll wait for your uploads, thanks.

**Reply by awohns** [reply]
Hi @clawsoon, the unified, inferred and dated tree sequences of the Thousand Genomes Project + Human Genome Diversity Project + Simons Genome Diversity project [are live on Zenodo](https://zenodo.org/record/5495535#.YTolVdNKhTY)! Let me know if you run into any issues with the data, I'm happy to help.
The tree sequences with high coverage ancients should be up soon as well.

**Reply by clawsoon** [reply]
Hi, @awohns - Great, thanks!  Three questions right off the bat:

 - What's the difference between the _p and _q tree sequences?
 - To triple check, is ts.node(i).time * 25 the correct way to get the age of node i in years?
 - Will the dates with ancients be different from/more accurate than the dates without ancients?

Thanks again!

**Reply by awohns** [reply]
> What's the difference between the _p and _q tree sequences?

These are the _p_ (shorter) and _q_ (longer) arms of each chromosome.

> To triple check, is ts.node(i).time * 25 the correct way to get the age of node i in years?

Yes that's right. This gives the _constrained_ node age estimate (details in Wohns et al. (2021) Methods). If you want the raw estimates from tsdate (in generations), the mean age estimate is `json.loads(ts.node(i).metadata)["mn"]` and the variance of the age estimate is `json.loads(ts.node(i).metadata)["vr"]`. 

> Will the dates with ancients be different from/more accurate than the dates without ancients?

It's hard to say: including the ancients "constrains" variant age estimates if a derived allele is observed in an ancient sample that is _older_ than the tsdate estimate of the node age. On the other hand, we didn't directly date the tree sequence with ancients; we used the age estimates from these tree sequences to build that tree sequence. So overall, date estimates should be similar between the tree sequences with and without ancients.

**Reply by clawsoon** [reply]
Thanks.
